qiskit/circuit/twirling.py

```diff
@@ -0,0 +1,132 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The twirling module."""
+
+from __future__ import annotations
+import typing
+
+from qiskit._accelerate.twirling import twirl_circuit as twirl_rs
+from qiskit.circuit.quantumcircuit import QuantumCircuit, _copy_metadata
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.library.standard_gates import CXGate, ECRGate, CZGate, iSwapGate
+from qiskit.exceptions import QiskitError
+
+if typing.TYPE_CHECKING:
+    from qiskit.transpiler.target import Target
+
+
+NAME_TO_CLASS = {
+    "cx": CXGate._standard_gate,
+    "ecr": ECRGate._standard_gate,
+    "cz": CZGate._standard_gate,
+    "iswap": iSwapGate._standard_gate,
+}
+
+
+def twirl_circuit(
+    circuit: QuantumCircuit,
+    twirling_gate: None | str | Gate | list[str] | list[Gate] = None,
+    seed: int | None = None,
+    num_twirls: int | None = None,
+    target: Target | None = None,
+) -> QuantumCircuit | list[QuantumCircuit]:
+    """Create a copy of a given circuit with Pauli twirling applied around a specified two qubit
+    gate.
+
+    Args:
+        circuit: The circuit to twirl
+        twirling_gate: The gate to twirl, defaults to `None` which means twirl all default gates:
+            :class:`.CXGate`, :class:`.CZGate`, :class:`.ECRGate`, and :class:`.iSwapGate`.
+            If supplied it can either be a single gate or a list of gates either as either a gate
+            object or it's string name. Currently only the names `"cx"`, `"cz"`, `"ecr"`,  and
+            `"iswap"` are supported. If a gate object is provided outside the default gates it must
+            have a matrix defined from it's :class:`~.Gate.to_matrix` method for the gate to potentially
```

Comment:
I'll naturally yield to english-native speakers, but shouldn't this be the following?
```suggestion
            object or its string name. Currently only the names `"cx"`, `"cz"`, `"ecr"`,  and
            `"iswap"` are supported. If a gate object is provided outside the default gates it must
            have a matrix defined from its :class:`~.Gate.to_matrix` method for the gate to potentially
```